python/pyiceberg/catalog/__init__.py

```diff
@@ -53,7 +54,15 @@ class Catalog(ABC):
 
     def __init__(self, name: str | None, **properties: str):
         self.name = name
-        self.properties = properties
+
+        if name is not None:
```

Comment:
`name` is required, so I don't think we need to worry about cases where it is `None`